uaclient/tests/test_gpg.py

```diff
@@ -0,0 +1,70 @@
+from uaclient import exceptions, gpg, util
+from uaclient.testing import data
+
+import logging
+import os
+import pytest
+
+
+class TestExportGPGKeyFromKeyring:
+
+    def test_key_error_on_missing_key(self, tmpdir):
+        """Raise UserFacingError when key_id is not in source_keyring_file."""
+        src_keyring = tmpdir.join('ubuntu-advantage-keyring')
+        src_keyring.write('')
+        destination_keyfile = tmpdir.join('destination_keyfile').strpath
+        key_id = 'Bueller... Bueller'
+        with pytest.raises(exceptions.UserFacingError) as excinfo:
+            gpg.export_gpg_key_from_keyring(
+                key_id=key_id, source_keyring_file=src_keyring.strpath,
+                destination_keyfile=destination_keyfile)
+
+        error_msg = "GPG key '%s' not found in %s" % (
+            key_id, src_keyring.strpath)
+        assert error_msg in str(excinfo.value)
+        assert False is os.path.exists(destination_keyfile)
+
+    @pytest.mark.parametrize('caplog_text', [logging.ERROR], indirect=True)
```

Comment:
Do we definitely need this line? I'd expect ERROR level messages to be included by default, and we do check the log level inline (by including the `ERROR    ` prefix in the string we look for in `caplog_text()`).

(It's really neither here nor there whether we include this for this test, but I would like to be sure that this isn't masking a bug in `caplog_text`.)

Reply:
dropped this